src/py21cmfast/src/HaloBox.c

```diff
@@ -329,32 +477,219 @@ int get_box_averages(double redshift, double norm_esc, double alpha_esc, double
+
+        double curr_vcb = consts->vcb_norel;
+        double M_turn_m = consts->mturn_m_nofb;
+        double M_turn_a = consts->mturn_a_nofb;
+        double M_turn_r;
+
+        float in_props[3];
+        float out_props[7];
+
+    #pragma omp for reduction(+:hm_avg,sm_avg,sm_avg_mini,sfr_avg,sfr_avg_mini,M_turn_a_avg,M_turn_m_avg,M_turn_r_avg,n_halos_cut)
+        for(i_halo=0; i_halo<halos->n_halos; i_halo++){
+            m = halos->halo_masses[i_halo];
+            //It is sometimes useful to make cuts to the halo catalogues before gridding.
+            //  We implement this in a simple way, if the user sets a halo's mass to zero we skip it
+            if(m == 0.){
+                n_halos_cut++;
+                continue;
+            }
+            x = halos->halo_coords[0+3*i_halo]; //NOTE:PerturbHaloField is on HII_DIM, HaloField is on DIM
+            y = halos->halo_coords[1+3*i_halo];
+            z = halos->halo_coords[2+3*i_halo];
+
+            //set values before reionisation feedback
+            //NOTE: I could easily apply reionization feedback without minihalos but this was not done previously
+            if(flag_options_stoc->USE_MINI_HALOS){
+                if(!flag_options_stoc->FIX_VCB_AVG && user_params_stoc->USE_RELATIVE_VELOCITIES)
+                    curr_vcb = ini_boxes->lowres_vcb[HII_R_INDEX(x,y,z)];
+
+                J21_val = redshift < global_params.Z_HEAT_MAX ? previous_spin_temp->J_21_LW_box[HII_R_INDEX(x,y,z)] : 0.;
+                Gamma12_val = redshift < global_params.Z_HEAT_MAX ? previous_ionize_box->Gamma12_box[HII_R_INDEX(x, y, z)] : 0.;
+                zre_val = redshift < global_params.Z_HEAT_MAX ? previous_ionize_box->z_re_box[HII_R_INDEX(x, y, z)] : 0.;
+                M_turn_m = lyman_werner_threshold(redshift, J21_val, curr_vcb, astro_params_stoc);
+                M_turn_r = reionization_feedback(redshift, Gamma12_val, zre_val);
+                M_turn_a = consts->mturn_a_nofb < M_turn_r ? M_turn_r : consts->mturn_a_nofb; //since ACG turnover not reassigned
+                if(M_turn_m < M_turn_r)M_turn_m = M_turn_r;
+            }
+
+            //these are the halo property RNG sequences
+            in_props[0] = halos->star_rng[i_halo];
+            in_props[1] = halos->sfr_rng[i_halo];
+            in_props[2] = halos->xray_rng[i_halo];
+
+            set_halo_properties(m,M_turn_a,M_turn_m,consts,in_props,out_props);
+
+            sfr = out_props[0];
+            sfr_mini = out_props[1];
+            nion = out_props[2];
+            wsfr = out_props[3];
+            stars = out_props[4];
+            stars_mini = out_props[5];
+            xray = out_props[6];
+
+            if(x+y+z == 0){
+                LOG_ULTRA_DEBUG("Cell 0 Halo %d: HM: %.2e SM: %.2e (%.2e) NI: %.2e SF: %.2e (%.2e) WS: %.2e",i_halo,m,stars,stars_mini,nion,sfr,sfr_mini,wsfr);
+                LOG_ULTRA_DEBUG("Mturn_a %.2e Mturn_m %.2e",M_turn_a,M_turn_m);
+            }
```

Comment:
again small thing, but since you have to do a computation to check whether you should log this, it'd be better if it was inside a pragma thing, like `#if LOG_LEVEL>=ULTRA_DEBUG`

Reply:
I would think any compiler should delete the line `if(x+y+z==0){}` but your way would be clearer